server/map_rpc_server.go

```diff
@@ -112,6 +112,23 @@ func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapL
 	return t.getLeavesByRevision(ctx, req.MapId, req.Index, mostRecentRevision)
 }
 
+// GetLeafByRevision returns an inclusion proof to either the leaf, or nil if the leaf does not exist.
+func (t *TrillianMapServer) GetLeafByRevision(ctx context.Context, req *trillian.GetMapLeafByRevisionRequest) (*trillian.GetMapLeafResponse, error) {
+	ctx, spanEnd := spanFor(ctx, "GetLeafByRevision")
+	defer spanEnd()
+	ret, err := t.getLeavesByRevision(ctx, req.MapId, [][]byte{req.Index}, mostRecentRevision)
+	if err != nil {
+		return nil, err
+	}
+	if got := len(ret.MapLeafInclusion); got != 0 {
```

Comment:
`got != 1` ? A test would have caught this.

Reply:
Sorry about that.  A gentleman always adds tests.